### DISCUSSION TITLE
ENOSPC: no space left on device, write

### DISCUSSION THREAD

**Discussion by pearcircuitmike** [vercel/community#831] [Help] [answered]
Hey, my build is failing halfway through deployment on Vercel with this error: ```Error occurred prerendering page "/global/79646". Read more: https://nextjs.org/docs/messages/prerender-error [ link was not helpful]
Error: ENOSPC: no space left on device, write"```

I found this issue but all the recent comments say it's resolved: https://github.com/vercel/vercel/issues/1664

Build is like 24gb on local. looks like im hitting some kind of vercel limit maybe? not sure which one though. didn't see anything on the website that I was exceeding. thanks!

**Comment by swarnava** [comment, marked as answer]
considering we give 20 gb space limit, 24 GB is still A LOT to have a frontend application. We will recommend to optimize your app and then try to deploy. If you have too many media files like videos, images, you can consider hosting it to a Storage service such as Amazon S3 to reduce the app size.

**Reply by pearcircuitmike** [reply]
Thanks! yes, it's a large site - 150k+ pages. I'll try to optimize to get it down. Thanks.
